### PR TITLE
webrtc: improve Firefox compability of RTCDTMFHelper

### PR DESCRIPTION
**Pull request by fippo**
does not check canInsertDTMF which is not implemented in Firefox yet.

### CODE REVIEW

File: webrtc/RTCDTMFSender-helper.js [hunk 1]
```diff
@@ -28,14 +28,17 @@ function createDtmfSender(pc = new RTCPeerConnection()) {
...
+    }
     // Wait until dtmfSender.canInsertDTMF becomes true.
     // Up to 150 ms has been observed in test. Wait 1 second
     // in steps of 10 ms.
     // Note: Using a short timeout and rejected promise in order to
     // make test return a clear error message on failure.
     return new Promise((resolve, reject) => {
       let counter = 0;
-      let checkfunc = function() {
+      step_timeout(function checkCanInsertDTMF() {
```
Comment (fippo) [inline webrtc/RTCDTMFSender-helper.js]: drive-by style change

File: webrtc/RTCDTMFSender-helper.js [hunk 2]
```diff
@@ -44,14 +47,11 @@ function createDtmfSender(pc = new RTCPeerConnection()) {
...
           ++counter;
-          step_timeout(checkfunc, 10);
+          step_timeout(checkCanInsertDTMF, 10);
         }
-      };
-      checkfunc();
+      }, 0);
     });
   }).then(() => {
-    assert_true(dtmfSender.canInsertDTMF,
```
Comment (fippo) [inline webrtc/RTCDTMFSender-helper.js]: this assertion used to be always true, the promise would have been rejected otherwise.

**Review by fippo — COMMENTED** [review]
@alvestrand PTAL. You don't want @jan-ivar to be sad about tests that fail when they could work ;-)

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25691)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/351200875)

**Review by alvestrand — APPROVED** [review]
Looks good. We need to test canInsertDtmf behavior too, but helpers shouldn't try to be tests.

File: webrtc/RTCDTMFSender-helper.js (hunk 1 above)
Comment (fippo) [reply webrtc/RTCDTMFSender-helper.js]: kept the busy-loop here, just changed the style ;-)
